Add EIP-5791: Physical Backed Tokens (#5791)
ethereum/EIPs: EIPS/eip-5791.md

---
eip: 5791
title: Physical Backed Tokens
description: Minimal interface for linking ownership of EIP-721 NFTs to a physical chip
author: 2pmflow (@2pmflow), locationtba (@locationtba), Cameron Robertson (@ccamrobertson), cygaar (@cygaar)
discussions-to: https://ethereum-magicians.org/t/physical-backed-tokens/11350
status: Draft
type: Standards Track
category: ERC
created: 2022-10-17
requires: 191, 721
---

## Abstract

> This standard is an extension of [EIP-721](./eip-721.md). It proposes a minimal interface for an [EIP-721](./eip-721.md) NFT to be "physically backed" and owned by whoever owns the NFT's physical counterpart.

Reviewer: ```suggestion
This standard is an extension of [EIP-721](./eip-721.md). It proposes a minimal interface for a [EIP-721](./eip-721.md) NFT to be "physically backed" and owned by whoever owns the NFT's physical counterpart.
```

## Motivation

NFT collectors enjoy collecting digital assets and sharing them with others online. However, there is currently no such standard for showcasing physical assets as NFTs with verified authenticity and ownership. Existing solutions are fragmented and tend to be susceptible to at least one of the following:

> - The NFT cannot proxy as ownership of the physical item. In most current implementations, the NFT and physical item are functionally two decoupled distinct assets after the NFT mint, in which the NFT can be freely traded independently from the physical item.

Reviewer: ```suggestion
- The ownership of the physical item and the ownership of the NFT are decoupled. 
```

> - Verification of authenticity of the physical item requires action from a trusted entity (e.g. StockX).

Reviewer: ```suggestion
- Verifying the authenticity of the physical item requires action from a trusted 3rd party (e.g. StockX).
```

> 
> This document proposes a solution to mitigate these issues in a decentralized way. Its intent is to be the simplest possible path towards linking physical items to digital NFTs without a centralized authority.

Reviewer: ```suggestion
```
Should be in the Rationale

## Specification

The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

### Requirements

> This approach assumes that the physical item must have a chip attached to it that fulfills the following requirements:

Reviewer: ```suggestion
This approach requres that the physical item must have a chip attached to it that fulfills the following requirements:
```

> - the ability to securely generate and store an ECDSA secp256k1 asymmetric keypair
> - the ability to sign messages from the private key of the asymmetric keypair
> - the ability for one to retrieve only the public key of the asymmetric keypair (private key non-extractable)

Reviewer: ```suggestion
- The chip can securely generate and store an ECDSA secp256k1 asymmetric keypair;
- The chip can sign messages using the private key of the previously-generated asymmetric keypair;
- The chip exposes the public key; and
- The private key cannot be extracted
```

    /// The implementation should also verify that the address signed in the signature matches msgSender.
    /// If the address recovered from the signature matches a chip address that's bound to an existing token, the token should be transferred to msgSender.
    /// If there is no existing token linked to the chip, the function should error.
    function transferTokenWithChip(
        bytes calldata signatureFromChip,
        uint256 blockNumberUsedInSig,
        bool useSafeTransferFrom
    ) external;

    /// @notice Calls transferTokenWithChip as defined above, with useSafeTransferFrom set to false.
    function transferTokenWithChip(bytes calldata signatureFromChip, uint256 blockNumberUsedInSig) external;

    /// @notice Emitted when a token is minted
    event PBTMint(uint256 indexed tokenId, address indexed chipAddress);

    /// @notice Emitted when a token is mapped to a different chip.
    /// Chip replacements may be useful in certain scenarios (e.g. chip defect).
    event PBTChipRemapping(uint256 indexed tokenId, address indexed oldChipAddress, address indexed newChipAddress);
}

```

> To aid recognition that an [EIP-721](./eip-721.md) token implements physical binding via this EIP: upon calling [EIP-721](./eip-721.md)’s `function supportsInterface(bytes4 interfaceID) external view returns (bool)` with `interfaceID=0xa70d2657`, a contract implementing this EIP must return true.

Reviewer: ```suggestion
To aid recognition that an [EIP-721](./eip-721.md) token implements physical binding via this EIP: upon calling [EIP-165](./eip-165.md)’s `function supportsInterface(bytes4 interfaceID) external view returns (bool)` with `interfaceID=0xa70d2657`, a contract implementing this EIP must return true.
```

The mint interface is up to the implementation. The minted NFT's owner should be the owner of the physical chip (this authentication could be implemented using the signature scheme defined for `transferTokenWithChip`).

## Rationale

The interface includes a `transferTokenWithChip` function that's opinionated with respect to the signature scheme, in order to enable a downstream aggregator-like product that supports transfers of any NFTs that implement this EIP in the future.

### Out of Scope

The following are some peripheral problems that are intentionally not within the scope of this EIP:

- trusting that a specific NFT collection's chip addresses actually map to physical chips embedded in items, instead of arbitrary EOAs
- ensuring that the chip does not deterioriate or get damaged
- ensuring that the chip stays attached to the physical item
- etc.

> Work is being done on these challenges in parallel. For example, the Ethereum Reality Service (`https://www.ers.to/`) is meant to tackle some of these issues.

Reviewer: ```suggestion
Work is being done on these challenges in parallel.
```

Author: Was also wondering what the downsides were of including this link. It's a relatively light landing page that mostly redirects to https://ethereum-magicians.org/t/a-proposal-for-the-ethereum-reality-service/9694. Would a link to ethereum-magicians directly be more appropriate?

Additionally, it's possible for the owner of the physical item to transfer the NFT to a wallet owned by somebody else (by sending a chip signature to that other person for use). We still consider the NFT physical backed, as ownership management is tied to the physical item. This can be interpreted as the item's owner temporarily lending the item to somebody else, since (1) the item's owner must be involved for this to happen as the one signing with the chip, and (2) the item's owner can reclaim ownership of the NFT at any time.

## Backwards Compatibility

This proposal is backward compatible with [EIP-721](./eip-721.md) on an API level. As mentioned above, for the token to be physical-backed, the contract must use a account-bound implementation of [EIP-721](./eip-721.md) (all [EIP-721](./eip-721.md) functions that transfer must throw) so that transfers go through the new function introduced here, which requires a chip signature.

## Reference Implementation

> At the time of writing, Kong is a supplier of chips that satisfy the cryptographic properties listed above.
> 

Reviewer: ```suggestion
```
